File: First Missing Positive.cpp
class Solution 
{
public:
    int firstMissingPositive(vector<int>& nums) 
    {
        int max = *max_element(nums.begin(), nums.end());
        unordered_map<int, bool> ex;

        for(int i = 0; i < nums.size(); i++)
        {
            ex[nums[i]] = true;
        }

        for(int i = 1; i < max; i++)
        {
            if(!ex[i])
            {
                return i; 
            }
        }

        if(max < 0)
        {
            return 1;
        }
        else
        {
            return max + 1;
        }
    }
};

File: First Missing Positive.py
class Solution:
    def firstMissingPositive(self, nums: List[int]) -> int:
        Max = max(nums)
        ex = {}

        for i in range(len(nums)):
            ex[nums[i]] = True

        for num in range(1, Max):
            if num not in ex:
                return num
        
        if Max < 0:
            return 1
        else: 
            return Max + 1